### PR TITLE
Updating repo

### PR DESCRIPTION
**Pull request by pulkitkejriwal1411**
This pull request resolves these issues.
https://github.com/Instadapp/dsa-connectors/issues/176
https://github.com/Instadapp/dsa-connectors/issues/129

- [x] Updated hardhat-etherscan latest-version i.e., [3.0.1](https://www.npmjs.com/package/@nomiclabs/hardhat-etherscan) in hardhat.config.ts.
- [x] Fix addLiquidity script. [here](https://github.com/Instadapp/dsa-connectors/blob/main/scripts/tests/addLiquidity.ts)
- [x] Upgrade test:runner script, enable usage of command line args to pass data.
- [ ] Upgrade solidity version of contract, maybe to 0.8.6.
- [ ] Explain any one connector test case in detail, as it will be helpful for understanding underlying code.

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -67,12 +67,12 @@ function getNetworkUrl(networkType: string) {
...
-  else if (networkType === "polygon") return POLYGONSCAN_API;
-  else if (networkType === "arbitrum") return ARBISCAN_API;
-  else return ETHERSCAN_API;
-}
+// function getScanApiKey(networkType: string) {
+//   if (networkType === "avalanche") return SNOWTRACE_API;
+//   else if (networkType === "polygon") return POLYGONSCAN_API;
+//   else if (networkType === "arbitrum") return ARBISCAN_API;
+//   else return ETHERSCAN_API;
+// }
```
Comment (pradyuman-verma) [inline hardhat.config.ts]: remove this

File: hardhat.config.ts [hunk 2]
```diff
@@ -123,7 +123,12 @@ const config: HardhatUserConfig = {
     tests: "./test",
   },
   etherscan: { 
-     apiKey: getScanApiKey(String(process.env.networkType)),
+     apiKey: {
+       mainnet: ETHERSCAN_API,
+       polygon: POLYGONSCAN_API,
+       arbitrumOne: ARBISCAN_API,
+       avalanche: SNOWTRACE_API
```
Comment (pradyuman-verma) [inline hardhat.config.ts]:
```suggestion
       mainnet: String(ETHERSCAN_API),
       polygon: String(POLYGONSCAN_API),
       arbitrumOne: String(ARBISCAN_API),
       avalanche: String(SNOWTRACE_API),
```

File: hardhat.config.ts (hunk 1 above)
Comment (pulkitkejriwal1411) [reply hardhat.config.ts]: Done

**Comment by pradyuman-verma** [comment]
Let's update this as well in the config:
```js
const networkGasPriceConfig: Record<string, string> = {
  "mainnet": "160",
  "polygon": "50",
  "avalanche": "50",
  "arbitrum": "2"
}
function createConfig(network: string) {
  return {
    url: getNetworkUrl(network),
    accounts: !!PRIVATE_KEY ? [`0x${PRIVATE_KEY}`] : { mnemonic },
    // gasPrice: 1000000, // 0.0001 GWEI
  };
}
```
Make gasPrice: networkGasPriceConfig[network] * 1e9;
And convert networkGasPriceConfig values into number.

File: hardhat.config.ts [hunk 3]
```diff
@@ -40,11 +40,11 @@ const mnemonic =
...
-const networkGasPriceConfig: Record<string, string> = {
-  "mainnet": "160",
-  "polygon": "50",
-  "avalanche": "50",
-  "arbitrum": "2"
+const networkGasPriceConfig: Record<string, Number> = {
+  "mainnet": 160000000000,
+  "polygon": 50000000000,
+  "avalanche": 50000000000,
+  "arbitrum": 2000000000
```
Comment (pradyuman-verma) [inline hardhat.config.ts]: Not like this, just do like:
```suggestion
  "mainnet": 160,
  "polygon": 50,
  "avalanche": 25,
  "arbitrum":1,
```

Comment (pulkitkejriwal1411) [reply hardhat.config.ts]: Resolved

File: scripts/tests/run-tests.ts [hunk 4]
```diff
@@ -4,32 +4,36 @@ import { promises as fs } from "fs";
...
+  .then(() =>
+    console.log(
+      `🙌 finished the test runner, time taken ${(end - start) / 1000} sec`
+    )
+  )
+  .catch((err) => console.error("❌ failed due to error: ", err));
+
+});
+
+async function testRunner(chain: string, testName: string) {
```
Comment (pradyuman-verma) [inline scripts/tests/run-tests.ts]: Let's pass them as optional args here.
```solidity
async function testRunner(chain?: string, testName?: string)
```

**Comment by pradyuman-verma** [comment]
Can you add the screenshot of both the test runner scripts?

**Comment by pulkitkejriwal1411** [comment]
Screenshot_20220305_210403


Screenshot of test:runner script

**Comment by pulkitkejriwal1411** [comment]
Screenshot_20220305_130302


Screenshot of runner:test script

**Comment by bhavik-m** [comment]
@pulkitkejriwal1411  why delete .env.example? can you restore it. it basically a reference to make your own env.

File: hardhat.config.ts [hunk 5]
```diff
@@ -115,15 +111,20 @@ const config: HardhatUserConfig = {
...
     sources: "./contracts",
     tests: "./test",
   },
   etherscan: { 
-     apiKey: getScanApiKey(String(process.env.networkType)),
+     apiKey: {
+       mainnet: String(ETHERSCAN_API),
+       polygon: String(POLYGONSCAN_API),
+       arbitrumOne: String(ARBISCAN_API),
+       avalanche: String(SNOWTRACE_API),
```
Comment (bhavik-m) [inline hardhat.config.ts:132]: @pulkitkejriwal1411 also add optimism and fantom api key

File: hardhat.config.ts [hunk 6]
```diff
@@ -40,18 +42,18 @@ const mnemonic =
...
-const networkGasPriceConfig: Record<string, string> = {
-  "mainnet": "160",
-  "polygon": "50",
-  "avalanche": "50",
-  "arbitrum": "2"
+const networkGasPriceConfig: Record<string, Number> = {
+  "mainnet": 160,
+  "polygon": 50,
+  "avalanche": 25,
+  "arbitrum":1,
```
Comment (bhavik-m) [inline hardhat.config.ts:52]: can you also add fantom and optimism here?

File: hardhat.config.ts (hunk 5 above)
Comment (pulkitkejriwal1411) [reply hardhat.config.ts:132]: Resolved.

**Review by bhavik-m — DISMISSED** [review]

File: hardhat.config.ts (hunk 6 above)
Comment (pulkitkejriwal1411) [reply hardhat.config.ts:52]: Added them. Please check if the prices are correct.

**Comment by Sowmayjain** [comment]
Hello, closing this PR due to high number of files changes with no proper documentation.

File: hardhat.config.ts (hunk 6 above)
Comment (bhavik-m) [reply hardhat.config.ts:52]: Yeah it's okay

**Pull request closed without merging**